edx_repo_tools/find_dependencies/find_python_dependencies.py

```diff
@@ -0,0 +1,82 @@
+"""
+Spider and catalog dependencies.
+$ python find_python_dependencies.py $FILE_PATH
+"""
+
+import click
+import json
+import os
+import requirements
+import sys
+from pathlib import Path
+import requests
+
+
+# The first of these we find is the requirements file we'll examine:
+def request_package_info_url(package):
+        base_url = "https://pypi.org/pypi/"
+        url = f"{base_url}{package}/json"
+        response = requests.get(url)
+        if response.status_code == 200:
+            data_dict = response.json()
+            info = data_dict["info"]
+            return info["home_page"]            
+        else:
+            print(f"Failed to retrieve data for package {package}. Status code:", response.status_code)    
+
+FIRST_PARTY_ORGS = ["openedx"]
+
+SECOND_PARTY_ORGS = [
+    "edx", "edx-unsupported", "edx-solutions",
+    "mitodl",
+    "overhangio",
+    "open-craft", "eduNEXT", "raccoongang",
+]
+
+def urls_in_orgs(urls, orgs):
+    """
+    Find urls that are in any of the `orgs`.
+    """
+    return sorted(
+        url for url in urls
+        if any(f"/{org}/" in url for org in orgs)
+    )
+
+@click.command()
+@click.option(
+    '--req-file', 'directories',
+    multiple=True,
+    help="The absolute file paths to locate Python dependencies" 
+        "within a particular repository. You can provide this "
+        "option multiple times to include multiple requirement files.",
+)
+@click.option(
+    '--ignore', 'ignore_paths',
+    multiple=True,
+    help="Path to ignore files that contain dependencies" 
+        "outside of your organization's approved list",
+)
+
+def main(directories=None, ignore_paths=None):
+    """
+    Analyze the requirements in input directory mentioned on the command line.    
+    """
+    packages_url = [] 
+
+    for directory in directories:
+        with open(directory) as fbase:
+            # Read each line (package name) in the file
+            for req in requirements.parse(fbase):
+                home_page = request_package_info_url(req.name)
+                if home_page is not None:
+                    if match := urls_in_orgs([home_page], SECOND_PARTY_ORGS):
+                        if home_page not in packages_url: 
```

Comment:
You could just make `home_page` a `set` and then you don't need to do this check.

Reply:
done